### ISSUE TITLE
Blob service calls fail when using a connection string

### ISSUE THREAD

**Issue by zooba** [Azure/azure-storage-python#143]
I grab my storage connection string from the portal and create my service with it, but then when I try to call anything I get this error:

```
>>> bs = azure.storage.blob.BlockBlobService(connection_string='...')
>>> bs.list_blobs('download')
Traceback (most recent call last):
  File "C:\Python35\lib\site-packages\azure\storage\storageclient.py", line 117, in _perform_request
    resp = self._filter(request)
  File "C:\Python35\lib\site-packages\azure\storage\storageclient.py", line 109, in _perform_request_worker
    return self._httpclient.perform_request(request)
  File "C:\Python35\lib\site-packages\azure\storage\_http\httpclient.py", line 222, in perform_request
    raise HTTPError(status, message, respheaders, respbody)
azure.storage._http.HTTPError: The requested URI does not represent any resource on the server.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python35\lib\site-packages\azure\storage\blob\baseblobservice.py", line 1154, in list_blobs
    resp = self._list_blobs(*args, **kwargs)
  File "C:\Python35\lib\site-packages\azure\storage\blob\baseblobservice.py", line 1224, in _list_blobs
    response = self._perform_request(request)
  File "C:\Python35\lib\site-packages\azure\storage\storageclient.py", line 125, in _perform_request
    _storage_error_handler(ex)
  File "C:\Python35\lib\site-packages\azure\storage\_serialization.py", line 130, in _storage_error_handler
    return _general_error_handler(http_error)
  File "C:\Python35\lib\site-packages\azure\storage\_error.py", line 71, in _general_error_handler
    raise AzureHttpError(message, http_error.status)
azure.common.AzureHttpError: The requested URI does not represent any resource on the server.
<?xml version="1.0" encoding="utf-8"?><Error><Code>InvalidUri</Code><Message>The requested URI does not represent any resource on the server.
RequestId:0e8b9823-0001-008d-79e2-7e4356000000
Time:2016-03-15T17:43:04.5101788Z</Message><UriPath>https://myaccount.blob.core.windows.net//download?restype=container&amp;comp=list</UriPath></Error>
```

Looking at the UriPath in the result, there seems to be an extra / before the container name, and indeed, if I remove the trailing slash from the endpoint specifier in the connection string, then it works. So guessing you need to either trim slashes from endpoints, or use a URL join function that doesn't add unnecessary slashes to URLs.

**Comment by emgerner-msft** [comment]
Hmmm, we tested with connection strings but I guess not in the same format you're using. Could you give us your connection string (with the sensitive parts xxx'ed out of course) so we can write a test and then fix?

**Comment by zooba** [comment]
`DefaultEndpointsProtocol=https;AccountName=ACCOUNTNAME;AccountKey=ACCOUNTKEY;BlobEndpoint=https://ACCOUNTNAME.blob.core.windows.net/;TableEndpoint=https://ACCOUNTNAME.table.core.windows.net/;QueueEndpoint=https://ACCOUNTNAME.queue.core.windows.net/;FileEndpoint=https://ACCOUNTNAME.file.core.windows.net/`

**Comment by emgerner-msft** [comment]
This is from the portal or something you constructed? When I tried getting the connection string from the portal I got something much shorter and more standard (which is what I'd expect): DefaultEndpointsProtocol=https;AccountName=ACCOUNTNAME;AccountKey=ACCOUNTKEY. 

We obviously should fix the issue on way or another, but it's preferable if portal doesn't return the endpoints so I want to investigate that as well. When I tried getting the connection I did so on a classic storage account, then Keys -> Primary Connection String. Did you do something different?

**Comment by zooba** [comment]
I got it from the portal:
![image](https://cloud.githubusercontent.com/assets/1693688/13789021/79cb7542-ea9f-11e5-84b5-18b189fc89cb.png)

It's a new style storage account, so perhaps that's the difference?

**Comment by emgerner-msft** [comment]
Verified, that's it. 

I'll fix the bug in the library this afternoon and follow up with the portal team offline to figure out why new vs old accounts is different.

**Comment by emgerner-msft** [comment]
Fixed as of 0.31.0.
